app/src/main/java/de/csicar/ning/Model.kt

```diff
@@ -39,19 +39,40 @@ data class DeviceWithName(
     val asDevice = Device(deviceId, networkId, ip, deviceName, hwAddress, isScanningDevice)
 
     val icon
-        get() = when (vendorName) {
-            "Espressif Inc." -> R.drawable.ic_memory_white_48dp
-            "ADMTEK INCORPORATED" -> R.drawable.ic_baseline_settings_ethernet_48
-            "LG Electronics (Mobile Communications)" -> R.drawable.ic_baseline_phone_android_48
-            "Nintendo Co.,Ltd" -> R.drawable.ic_baseline_videogame_asset_48
-            "AzureWave Technology Inc." -> R.drawable.ic_baseline_cast_48
-            "Compal Broadband Networks, Inc." -> R.drawable.ic_baseline_router_48
-            "Ubiquiti Networks Inc." -> R.drawable.ic_baseline_router_48
-            "Sonos, Inc." -> R.drawable.ic_baseline_speaker_48
-            "AVM Audiovisuelles Marketing und Computersysteme GmbH" -> R.drawable.ic_baseline_router_48
-            "TP-LINK TECHNOLOGIES CO.,LTD." -> R.drawable.ic_baseline_router_48
-            "HUAWEI TECHNOLOGIES CO.,LTD" -> R.drawable.ic_baseline_phone_android_48
-            "Xiaomi Communications Co Ltd" -> R.drawable.ic_baseline_phone_android_48
+        get() = when {
+            isScanningDevice -> R.drawable.ic_baseline_phone_android_48
+
+            /**
+             * Icons based on MAC address
+             */
+            hwAddress?.address?.contains("52:54:00") == true -> R.drawable.ic_laptop_white_48dp // Unofficial mac address range used by KVM
```

Comment:
A new icon for virtual machines is required here, but I didn't find a suitable one.

Reply:
Using the MAC-Rage of KVMs is a great idea!

Maybe the `layers` icon could do the trick? https://material.io/resources/icons/?icon=layers&style=baseline

If you don't feel comfortable with adding a new icon, let me. Then I'll to that for you :)

---

BTW: I'd go for `hwAddress?.addres?.startsWith("52:54:00") ->` instead of `.contains`. That way we don't accidentally classify devices with the "52:54:00" suffix as a VM

Reply:
`layers` might work, but maybe there's a better match. I usually take icons from https://materialdesignicons.com/. If none from there suits in this case, we could ask them to create an icon.

Reply:
That's a good idea. License also seems to be compatible: https://github.com/Templarian/MaterialDesign/blob/master/LICENSE

Reply:
For now I took the `layers` icon,